server/src/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1790,6 +1802,10 @@
             sc.setParameters("display", display);
         }
 
+        if (ids != null && !ids.isEmpty()) {
```

Comment:
What about extracting line 1805 - 1808 to a method. Those lines are the same as the ones at 3172-3175. then, you could also write test cases and add java doc for it.